cli/parse-options.js

```diff
@@ -34,8 +38,27 @@ function parseOptions (options) {
     config.url = options.url
   }
 
+  PouchDB.plugin(require('pouchdb-mapreduce'))
+
   if (options.dbUrl) {
-    config.db.url = options.dbUrl
+    PouchDB.plugin(require('pouchdb-adapter-http'))
+    dbOptions.prefix = options.dbUrl
+
+    if (options.inMemory) {
```

Comment:
this should not be within the `if (options.dbUrl) {` block. If `options.dbUrl` is it means that the data will be written to a CouchDB at  the URL set in `options.dbUrl`, the `options.inMemory` should be ignored at this point.

And the `config.PouchDB = PouchDB.default(dbOptions)` must be run for all cases, so move it out of the `if` blocks.

What you want is something like this:

```js
if (options.dbUrl) {
  PouchDB.plugin(require('pouchdb-adapter-http'))
  dbOptions.prefix = options.dbUrl
} else {
  if (options.inMemory) {
    PouchDB.plugin(require('pouchdb-adapter-memory'))
    log.info('config', 'Storing all data in memory only')
  } else {
    PouchDB.plugin(require('pouchdb-adapter-leveldb'))
    dbOptions.prefix = config.paths.data || '.hoodie'
  }
}

config.PouchDB = PouchDB.default(dbOptions)
```

Does that make sense?